File: app/src/main/java/com/magenta/rx/java/http/TranslateClient.java
package com.magenta.rx.java.http;

import com.magenta.rx.kotlin.record.TranslateAnswer;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface TranslateClient {

    @GET(value = "/api/v1.5/tr.json/translate")
    Observable<TranslateAnswer> translate(@Query("key") String key, @Query("text") String text, @Query("lang") String lang);
}

File: app/src/main/kotlin/com/magenta/rx/kotlin/http/TranslateClient.kt
package com.magenta.rx.kotlin.http

import com.magenta.rx.kotlin.record.TranslateAnswer
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface TranslateClient {

    @GET("/api/v1.5/tr.json/translate") fun translate(@Query("key") key: String, @Query("text") text: String, @Query("lang") lang: String): Observable<TranslateAnswer>
}